### PR TITLE
Creating message notifications

### PR DESCRIPTION
**Pull request by ShadowSpin1**
I demo'd what i made :D

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -15,6 +15,12 @@ def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:account_update, keys: %i[first_name last_name weight height belt years_of_experience address description gender photo training_style])
   end
 
+  def set_notifications
```
Comment (ShadowSpin1) [inline app/controllers/application_controller.rb:18]: This is for universal access of notifications so I can add purple circles on notification in footer.

File: app/controllers/chatrooms_controller.rb [hunk 2]
```diff
@@ -2,6 +2,7 @@ class ChatroomsController < ApplicationController
   def show
     @chatroom = Chatroom.find(params[:id])
     @message = Message.new
+    set_notifications_to_read
```
Comment (ShadowSpin1) [inline app/controllers/chatrooms_controller.rb:5]: private method to set notifications to read status.

File: app/controllers/chatrooms_controller.rb [hunk 3]
```diff
@@ -22,4 +23,9 @@ def my_messages
   def chatroom_params
     params.permit(:name)
   end
+
```
Comment (ShadowSpin1) [inline app/controllers/chatrooms_controller.rb:26]: when you open a given chatroom, you will set the notifications for that specific chatroom from unread status to read.

File: app/controllers/messages_controller.rb [hunk 4]
```diff
@@ -8,8 +8,9 @@ def create
     if @message.save
       ChatroomChannel.broadcast_to(
         @chatroom,
-        render_to_string(partial: "message", locals: {message: @message})
+        render_to_string(partial: "message", locals: { message: @message })
```
Comment (ShadowSpin1) [inline app/controllers/messages_controller.rb:11]: added space between the {}

File: app/controllers/messages_controller.rb [hunk 5]
```diff
@@ -8,8 +8,9 @@ def create
...
       )
+      notify_recipient
```
Comment (ShadowSpin1) [inline app/controllers/messages_controller.rb:13]: private method to notify recipient

File: app/controllers/messages_controller.rb [hunk 6]
```diff
@@ -21,4 +22,15 @@ def create
   def message_params
     params.require(:message).permit(:content)
   end
+
+  def notify_recipient
+    users_in_room = @message.joined_users
```
Comment (ShadowSpin1) [inline app/controllers/messages_controller.rb:27]: takes users of chatroom, and sends notification only to recipient (not sender of course)

File: app/controllers/messages_controller.rb [hunk 7]
```diff
@@ -21,4 +22,15 @@ def create
...
+    users_in_room.each do |user|
+      next if user.eql?(current_user)
+
+      notification = MessageNotification.with(message: @message.content, chatroom: @message.chatroom)
```
Comment (ShadowSpin1) [inline app/controllers/messages_controller.rb:31]: creates Notification.

File: app/notifications/message_notification.rb [hunk 8]
```diff
@@ -6,10 +6,15 @@
 class MessageNotification < Noticed::Base
   # Add your delivery methods
   #
-  # deliver_by :database
-  # deliver_by :email, mailer: "UserMailer"
-  # deliver_by :slack
-  # deliver_by :custom, class: "MyDeliveryMethod"
+  deliver_by :database
```
Comment (ShadowSpin1) [inline app/notifications/message_notification.rb:9]: this adds notifs to DB

File: app/notifications/message_notification.rb [hunk 9]
```diff
@@ -6,10 +6,15 @@
...
+  deliver_by :action_cable, channel: ChatroomChannel, stream: :custom_stream, format: :action_cable_data
```
Comment (ShadowSpin1) [inline app/notifications/message_notification.rb:10]: this allows notifs to be sent real time

File: app/views/chatrooms/_chat.html.erb [hunk 10]
```diff
@@ -24,6 +24,7 @@ if last_message
       </div>
       <div>
         <small class="text-nowrap"><%= last_message.created_at.strftime("%b %e") %></small>
+        <%= "You have a new message!" unless chat.notifications_as_chatroom.where(recipient: current_user, read_at: nil).empty? %>
```
Comment (ShadowSpin1) [inline app/views/chatrooms/_chat.html.erb:27]: show view logic for notification appearance only when notifications are unread.

File: app/controllers/messages_controller.rb (hunk 7 above)
Comment (jon-troupe) [reply app/controllers/messages_controller.rb:31]: Looks like it also skips notifying if it's from the current user?

**Review by laura-gonzalez-z — APPROVED** [review]

**Pull request merged**